Item #8853: Ontology, Form insert and update

Sigmonia opened the pull request:
#### Rationale
Item #8853: Ontology, Form insert and update

#### Related Pull Requests
* https://github.com/LabKey/ontology/pull/43

#### Changes
* Adds the ConceptSubtree DomainProperty so we can apply a specific ontology hierarchy path to a field

labkey-matthewb committed:
ConceptSubtree column property -- checkpoint
(still need schema change)

labkey-matthewb committed:
ConceptSubtree column property -- checkpoint
(still need schema change)

Sigmonia committed:
Use column caption in filterdialog for concept browser link

Sigmonia committed:
Use column caption in filterdialog for concept browser link

labkey-matthewb committed:
ConceptSubtree column property
use to reduce synonym ambiguity on import

labkey-matthewb committed:
ConceptSubtree column property
use to reduce synonym ambiguity on import

labkey-matthewb committed:
new column PropertyDescriptor.ConceptSubtree

labkey-matthewb committed:
new column PropertyDescriptor.ConceptSubtree

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_conceptdisplaycolumn

labkey-matthewb committed:
merge fixup

labkey-matthewb committed:
merge fixup

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_conceptdisplaycolumn

Sigmonia committed:
Ontology changes to support showing the Concept Picker for the Insert/update pages
- Add ConceptInput React entrypoint
- Loading JSP
- Alpha components package bump
- Initial layout of React elements and styling

Sigmonia committed:
Alpha bump .2

Sigmonia committed:
Move ConceptInput app to the Ontology module

Sigmonia committed:
Fix issue where FilterDialog input doesn't get updated

cnathe committed:
Merge branch 'develop' into fb_conceptdisplaycolumn

cnathe commented on api/src/org/labkey/api/data/ColumnRenderPropertiesImpl.java:
```diff
@@ -865,64 +867,79 @@ public void setPrecision(int precision)
 
     @Override
-    public void setConceptLabelColumn(String conceptLabelColumn)
+    public void setSourceOntology(String sourceOntology)
```
I noticed that several of these ontology related setters don't have the `assert _checkLocked();` like the others. Not likely part of this PR, but is that intentional for any reason that you know of? or should they be added?

cnathe commented on api/src/org/labkey/api/exp/OntologyManager.java:
```diff
@@ -1564,15 +1564,47 @@ private static PropertyDescriptor ensurePropertyDescriptor(PropertyDescriptor pd
+        }
+
+
+/*        TableInfo t = get();
```
There is a lot of commented out code left here. Is that to be removed at this point?

cnathe commented on experiment/resources/schemas/dbscripts/sqlserver/exp-21.004-21.005.sql:
```diff
@@ -0,0 +1 @@
+--ALTER TABLE exp.PropertyDescriptor ADD ConceptSubtree NVARCHAR(MAX) NULL;
```
I assume we do want to support this in sql server. Is there a reason this is commented out?

cnathe reviewed (changes requested):
I reviewed the changes and made a few comments, but @labkey-matthewb I think it would be good if you took a look at the changes for `OntologyManager.java` and the related classes. I am not familiar with that code.

cnathe commented:
> I reviewed the changes and made a few comments, but @labkey-matthewb I think it would be good if you took a look at the changes for `OntologyManager.java` and the related classes. I am not familiar with that code.

Looking at the commits, maybe you were the one that made those changes @labkey-matthewb. Is that correct?

labkey-matthewb committed:
Merge branch 'fb_conceptdisplaycolumn' of https://github.com/LabKey/platform into fb_conceptdisplaycolumn

labkey-matthewb committed:
_checkLocked

labkey-matthewb committed:
remove comment

labkey-matthewb committed:
remove commented code

Sigmonia committed:
bump alpha version to .3

Sigmonia committed:
bump alpha version to .4

Sigmonia committed:
Merge remote-tracking branch 'origin/develop' into fb_conceptdisplaycolumn

Sigmonia committed:
merge develop into fb

Sigmonia committed:
bump alpha version after merge, .5

Sigmonia commented on api/src/org/labkey/api/exp/OntologyManager.java:
```diff
@@ -2355,88 +2271,95 @@ public static PropertyDescriptor insertOrUpdatePropertyDescriptor(PropertyDescri
     static final String parameters = "propertyuri,name,description,rangeuri,concepturi,label," +
```
@mbellew There are some fields that are no longer being included in the insert statement. Should the be added here?

Sigmonia replied:
facetingbehaviortype,
phi,
redactedText,
excludefromshifting,
mvindicatorstoragecolumnname,
defaultscale,

labkey-matthewb replied:
This was an attempt to make the code _more_ consistent instead of repeating the columns in three places.  I didn't realize that insertPropertyIfNotExists() handled more columns than insertPropertyDescriptors().  I can't imagine that's intentional.  I assume that's genetic drift?

Sigmonia committed:
Fix issue where field type wasn't getting set correctly

Sigmonia committed:
include additional missing fields

Sigmonia committed:
bump alpha after components merge

Sigmonia committed:
merge develop into fb

Sigmonia committed:
Bump alpha to .6

Sigmonia committed:
Add ontology abbreviation to ConceptModel and Concept java interface serialization
Use ConceptModel's new ontology prop to set ConceptAnnotation's modal source ontology
fix jest tests
Adjust annotation concept display to default to principal code
Adjust OnotologyLookup domainproperty field change calculation to include field type check

Sigmonia committed:
Bump alpha to .7

Sigmonia committed:
merge develop into fb

Sigmonia committed:
fix possible NPE

Sigmonia committed:
Add conceptsubtree to schema xml

labkey-matthewb committed:
Helper code for non-DataIterator handling of concept lookup columns.
This handles code->label mapping.  Does not yet handle import column.

Sigmonia committed:
Bump alpha to .8

Sigmonia committed:
Merge remote-tracking branch 'origin/develop' into fb_conceptdisplaycolumn

Sigmonia committed:
Bump alpha after components merge

Sigmonia committed:
Close leaked connection

Sigmonia committed:
merge develop into fb
- Bumped fb sql script versions due to merged scripts.

Sigmonia committed:
Fix up some of the new annotation tests and timings

Sigmonia committed:
Merge remote-tracking branch 'origin/develop' into fb_conceptdisplaycolumn

Sigmonia committed:
Bump alpha after dev merge .9

Sigmonia committed:
Finalize components package version number
